### PR TITLE
feat: remove encryption upon registering second signature

### LINKED ISSUE #505: [Second Signature] detect and remove wallet encryption during second signature registration
If a second signature registration is initiated the wallet should check if the current wallet has been imported using an encryption password. If that is the case, a modal needs to be displayed, informing the user that the encryption password will be removed after the second signature registration has been broadcast successfully.

Important:

- if an encryption password is present, that password will still be used to authorize the transaction
- the encryption password is removed after the transaction has been broadcast successfully, the wallet should not wait for the confirmation of said transaction

### PR DESCRIPTION
**Pull request by dated**
## Summary

Closes #505.

## Checklist

-   [ ] My changes look good in both light AND dark mode
-   [ ] The change is not hardcoded to a single network, but has multi-asset in mind
-   [ ] I checked my changes for obvious issues, debug statements and commented code
-   [ ] Documentation _(if necessary)_
-   [x] Tests _(if necessary)_
-   [x] Ready to be merged

### CODE REVIEW

**Review by faustbrian — DISMISSED** [review]

**Review by samharperpittam — CHANGES_REQUESTED** [review]
@dated  - This doesn't appear to be removing the Enc Password after registering a second sig.

Trying to send a transaction after registering a second sig still asks for a Enc Password and results in the following error:

```
{"message":"This wallet does not use PBKDF2 encryption.","type":"Error"}
``

File: src/domains/dashboard/components/Wallets/Wallets.tsx [hunk 1]
```diff
@@ -144,12 +144,9 @@ export const Wallets: FC<WalletsProperties> = ({
 				/>
 			)}
 
-			<DeleteWallet
-				isOpen={modal === "delete"}
-				onClose={resetWalletAction}
-				onCancel={resetWalletAction}
-				onDelete={handleDeleteWallet}
-			/>
+			{modal === "delete" && (
```
Comment (ignromanov) [inline src/domains/dashboard/components/Wallets/Wallets.tsx:147]: Would be good to implement easy types for the `modal` local state value, like:
`export type WalletModalTypes =  "rename" | "delete";`

**Comment by faustbrian** [comment]
@dated conflicts

**Comment by dated** [comment]
> @dated conflicts

done.

**Review by samharperpittam — APPROVED** [review]

**Pull request merged**